src/omv/ports/stm32/jpeg.c

```diff
@@ -744,6 +751,9 @@ void jpeg_decompress(image_t *dst, image_t *src) {
 
     if ((JPEG_state.jpeg_descr.Conf.ColorSpace == JPEG_YCBCR_COLORSPACE) && dst->is_color) {
         HAL_DMA2D_DeInit(&DMA2D_Handle);
+
+        // Ensure any cached reads are dropped.
+        SCB_InvalidateDCache_by_Addr((uint32_t *) dst->data, image_size(dst));
```

Comment:
The whole dst image is invalidated right after configuring the DMA2D: https://github.com/openmv/openmv/blob/19d70d69e887cc64f67df653cc02f81db68be911/src/omv/ports/stm32/jpeg.c#L506-L510

Then every row/block is invalidated before the transfer: https://github.com/openmv/openmv/blob/19d70d69e887cc64f67df653cc02f81db68be911/src/omv/ports/stm32/jpeg.c#L712-L714

Now here `dst` is invalidated again after all transfers are complete.  The first invalidate is definitely Not needed, as every block will be invalidated anyway before the transfer, or all 3 of them can be replaced with just one invalidate after the `DMA2D_PollForTransfer`.

Reply:
Yeah, that's why I dropped `SCB_InvalidateDCache_by_Addr((uint32_t *) rp, dst->w * dy * sizeof(uint16_t));` in the PR given the whole image is already invalidated before starting DMA2D.

> Also possible to replace all 3 of them with just one invalidate before/after the DMA2D_PollForTransfer.

Yeah, you could, there's less overhead doing them all in one go though before and after than before and after every row though right?

Reply:
> before and after everything than before and after every row, right?

I think so.

Reply:
Okay, then this PR is ready.

Reply:
BTW if you want to save some time and optimize this, you could try invalidating the previous line/block *before* the poll.

Reply:
That is invalidate while the DMA is transferring the next line/block.

Reply:
Implemented the suggestion. Works.